docs/exposing-binding-data/olm-descriptors.md

```diff
@@ -0,0 +1,184 @@
+resource, use StatusDescriptors.
+
+The secret generation behavior explained in the annotations is applicable for
+OLM descriptors also.  The data model is also same as that of annotations, but
+the syntax is different to match OLM descriptors naming pattern.
+
+The only two fields that is used for binding is `Path` and `X-Descriptors`.
+Path is a dot-delimited path of the field on the object that the descriptor
+describes. The X-Descriptors define the binding meta-data similar to CR/CRD
+annotation.
+
+If the path is pointing to a `Secret` resource, there should be an X-Descriptors like this:
+
+    urn:alm:descriptor:io.kubernetes:Secret
+
+Similary, if the path is pointing to a `ConfigMap` resource, there should be an X-Descriptors like this:
+
+    urn:alm:descriptor:io.kubernetes:ConfigMap
+
+If the `Secret` or `ConfigMap` specific X-Descriptors are not present, that
+descriptor should be accessing multiple values at the given path.
+
+There should be a `service.binding` entry in the X-Descriptors to identify this
+is a Service Binding configguration.
+
+## Mount an entire Secret as the binding Secret
+
+One of the common use case is to inject all the values from a Secret resource.
+The Secret resource must be specified as an attribute in the custom resource.
+If the attribute is part of `.spec`, you can create SpecDescriptor, and if it's
+part of the `.status`, you can create StatusDescriptor.  In the descriptor, you
+can use `path` attribute to point to field with the Secret resource name.
+
+Here is an example configuration:
+
+```
+- path: data.dbCredentials
+  x-descriptors:
+  - urn:alm:descriptor:io.kubernetes:Secret
+  - service.binding
+```
+
+The X-Descriptor has `urn:alm:descriptor:io.kubernetes:Secret` entry to indicate
+the path is pointing to a Secret resource.
+
+## Mount an entire ConfigMap as the binding Secret
+
+Another common use case is to inject all the values from a ConfigMap resource.
+The ConfigMap resource must be specified as an attribute in the custom resource.
+If the attribute is part of `.spec`, you can create SpecDescriptor, and if it's
+part of the `.status`, you can create StatusDescriptor.  In the descriptor, you
+can use `path` attribute to point to field with the ConfigMap resource name.
+
+Here is an example configuration:
+
+```
+- path: data.dbConfiguration
+  x-descriptors:
+  - urn:alm:descriptor:io.kubernetes:ConfigMap
+  - service.binding
+```
+
+The X-Descriptor has `urn:alm:descriptor:io.kubernetes:ConfigMap` entry to
+indicate the path is pointing to a ConfigMap resource.
+
+## Mount an entry from a ConfigMap/Secret into the binding Secret
+
+To specify a particular entry from a Secret or ConfigMap, the X-Descriptor can
+update `service.binding` line with a name and `sourceKey`.  Here is an example:
+
+```
+- path: data.dbConfiguration
+  x-descriptors:
+  - urn:alm:descriptor:io.kubernetes:ConfigMap
+  - service.binding:my_certificate:sourceKey=certificate
```

Comment:
is this possible with multiple entries from a configmap/secret?

Reply:
Separate x-descriptors will be required for multiple entries.